tests/utils.py

```diff
@@ -55,14 +56,17 @@ def __init__(
 def create_server(
     server_thread_class: type[_T_co],
 ) -> Callable[[type[BaseRequestHandler]], AbstractContextManager[_T_co]]:
+    port_counter = AtomicInteger(7777)
+
     @contextlib.contextmanager
     def server(handler_class: type[BaseRequestHandler]) -> Generator[_T_co, None, None]:
         lock = filelock.FileLock(LOCK_PATH)
         with lock:
-            server_thread = server_thread_class(handler_class, daemon=True)
+            port = port_counter.add_and_get(1)
+            server_thread = server_thread_class(handler_class, port, daemon=True)
             server_thread.start()
             try:
-                yield server_thread
+                yield port
```

Comment:
Is it a bug that this change doesn't fail `mypy` type checks?  (re: the `_T_co` typevar)

Reply:
The `atomos` library is not typed so `add_and_get(1)` returns `Any`